snagsby/sources.py

```diff
@@ -63,7 +58,54 @@ def options(self):
 
     @property
     def region_name(self):
-        return self.options.get('region')
+        return self.options.get('region', 'us-west-2')
+
+    def get_data(self):
+        return sanitize(self.get_raw_data())
+
+class SMSource(AWSSource):
+    def get_sm_response(self):
+        key = "{}{}".format(self.url.netloc, self.url.path)
+        session = boto3.session.Session()
+        endpoint_url = "https://secretsmanager.{}.amazonaws.com".format(self.region_name)
+        client = session.client(
+            service_name='secretsmanager',
+            region_name=self.region_name,
+            endpoint_url=endpoint_url
+        )
+
+        try:
+            return client.get_secret_value( SecretId=key )
+        except botocore.exceptions.ClientError as e:
+            if e.response['Error']['Code'] == 'ResourceNotFoundException':
+                logger.debug("The requested secret " + key + " was not found")
+            elif e.response['Error']['Code'] == 'InvalidRequestException':
+                logger.error("The request was invalid due to:", e)
+            elif e.response['Error']['Code'] == 'InvalidParameterException':
+                logger.error("The request had invalid params:", e)
+            return {}
+
+    def get_raw_data(self):
```

Comment:
Nice. Let's write some tests around json decoding errors and missing `SecureString`

Reply:
Done